app/src/main/java/com/github/se/orator/ui/offline/OfflineRecordingScreen.kt

```diff
@@ -109,30 +109,62 @@ fun OfflineRecordingScreen(
             horizontalAlignment = Alignment.CenterHorizontally) {
               Box(
                   contentAlignment = Alignment.Center,
-                  modifier =
-                      Modifier.size(AppDimensions.logoSize)
-                          .testTag("MicIconContainer")) { // // should be 203.dp
-                    Image(
-                        painter = painterResource(id = R.drawable.bckgrnd_blobs),
-                        contentDescription = "Background",
-                        modifier = Modifier.size(AppDimensions.logoSize).testTag("BackgroundBlob"))
-                    Icon(
-                        imageVector = Icons.Filled.Mic,
-                        contentDescription = "Microphone",
-                        modifier =
-                            Modifier.size(AppDimensions.iconSizeMic)
-                                .scale(if (isRecording) scale else 1f)
-                                .clickable {
-                                  if (permissionGranted) {
-                                    speakingViewModel.onMicButtonClicked(true)
-                                  } else {
-                                    permissionLauncher.launch(Manifest.permission.RECORD_AUDIO)
-                                  }
+                  modifier = Modifier.size(AppDimensions.logoSize).testTag("MicIconContainer")) {
+                    MicrophoneButton(
+                        viewModel,
+                        collState,
+                        permissionGranted,
+                        context,
+                        funRec = {
+                          // what to do when user begins to record a file
+                          if (analysisState.value == SpeakingRepository.AnalysisState.IDLE ||
+                              analysisState.value == SpeakingRepository.AnalysisState.IDLE) {
```

Comment:
It seems that the condition analysisState.value == SpeakingRepository.AnalysisState.IDLE is being evaluated twice. To keep the code clean and efficient, this can be simplified to a single check instead of repeating the same condition.

Reply:
fixed